crates/driver/src/domain/competition/mod.rs

```diff
@@ -386,10 +386,13 @@ pub struct Solved {
     pub gas: Option<eth::Gas>,
 }
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct Amounts {
-    pub sell: eth::TokenAmount,
-    pub buy: eth::TokenAmount,
+    pub side: order::Side,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub traded_sell: eth::TokenAmount,
+    pub traded_buy: eth::TokenAmount,
```

Comment:
This needs serious docs (how is our next hire expected to guess that `sell.amount` is supposed to be the limit price)?

I'm also not sure that using Asset in one place and TokenAmount in the other is a great choice (but I wouldn't feel strongly if it was clear from the names/docs on what is what).

Reply:
added comments